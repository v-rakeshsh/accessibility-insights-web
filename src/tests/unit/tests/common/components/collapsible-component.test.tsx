// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import * as React from 'react';

import {
    CollapsibleComponent,
    CollapsibleComponentProps,
} from '../../../../../common/components/collapsible-component';
import {
    expectMockedComponentPropsToMatchSnapshots,
    mockReactComponents,
} from '../../../mock-helpers/mock-module-helpers';

jest.mock('@fluentui/react-components');
describe('CollapsibleComponentTest', () => {
    mockReactComponents([Button]);
    test('render expanded with content-class-name', () => {
        const props: CollapsibleComponentProps = {
            header: <div>Some header</div>,
            content: <div>Some content</div>,
            contentClassName: 'content-class-name',
        };
        const renderResult = render(<CollapsibleComponent {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });

    test('render expanded without content-class-name', () => {
        const props: CollapsibleComponentProps = {
            header: <div>Some header</div>,
            content: <div>Some content</div>,
        };
        const renderResult = render(<CollapsibleComponent {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });

    test('render with container-class-name', () => {
        const props: CollapsibleComponentProps = {
            header: <div>Some header</div>,
            content: <div>Some content</div>,
            containerClassName: 'a-container',
        };

        const renderResult = render(<CollapsibleComponent {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });

    test('render without container-class-name', () => {
        const props: CollapsibleComponentProps = {
            header: <div>Some header</div>,
            content: <div>Some content</div>,
        };

        const renderResult = render(<CollapsibleComponent {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });

    test('toggle from expanded to collapsed', async () => {
        const props: CollapsibleComponentProps = {
            header: <div>Some header</div>,
            content: <div>Some content</div>,
        };

        const renderResult = render(<CollapsibleComponent {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot('expanded');
        expectMockedComponentPropsToMatchSnapshots([Button]);
        await userEvent.click(renderResult.container.querySelector('mock-customizedactionbutton'));
        expect(renderResult.asFragment()).toMatchSnapshot('collapsed');
        expectMockedComponentPropsToMatchSnapshots([Button]);
    });
});
