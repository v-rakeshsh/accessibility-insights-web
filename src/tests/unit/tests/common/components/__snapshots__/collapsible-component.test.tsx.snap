// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleComponentTest render expanded with content-class-name 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <mock-button
      appearance="transparent"
      aria-expanded="true"
      classname="collapsible"
      icon="[object Object]"
    >
      <div>
        Some header
      </div>
    </mock-button>
    <div
      class="content-class-name collapsibleContent"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleComponentTest render expanded with content-class-name: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-expanded": true,
  "children": <div>
    Some header
  </div>,
  "className": "collapsible",
  "icon": <ChevronDownRegular
    className="collapsible"
  />,
  "onClick": [Function],
}
`;

exports[`CollapsibleComponentTest render expanded without content-class-name 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <mock-button
      appearance="transparent"
      aria-expanded="true"
      classname="collapsible"
      icon="[object Object]"
    >
      <div>
        Some header
      </div>
    </mock-button>
    <div
      class="collapsibleContent"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleComponentTest render expanded without content-class-name: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-expanded": true,
  "children": <div>
    Some header
  </div>,
  "className": "collapsible",
  "icon": <ChevronDownRegular
    className="collapsible"
  />,
  "onClick": [Function],
}
`;

exports[`CollapsibleComponentTest render with container-class-name 1`] = `
<DocumentFragment>
  <div
    class="a-container"
  >
    <mock-button
      appearance="transparent"
      aria-expanded="true"
      classname="collapsible"
      icon="[object Object]"
    >
      <div>
        Some header
      </div>
    </mock-button>
    <div
      class="collapsibleContent"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleComponentTest render with container-class-name: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-expanded": true,
  "children": <div>
    Some header
  </div>,
  "className": "collapsible",
  "icon": <ChevronDownRegular
    className="collapsible"
  />,
  "onClick": [Function],
}
`;

exports[`CollapsibleComponentTest render without container-class-name 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <mock-button
      appearance="transparent"
      aria-expanded="true"
      classname="collapsible"
      icon="[object Object]"
    >
      <div>
        Some header
      </div>
    </mock-button>
    <div
      class="collapsibleContent"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleComponentTest render without container-class-name: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-expanded": true,
  "children": <div>
    Some header
  </div>,
  "className": "collapsible",
  "icon": <ChevronDownRegular
    className="collapsible"
  />,
  "onClick": [Function],
}
`;

exports[`CollapsibleComponentTest toggle from expanded to collapsed: Button props 1`] = `
{
  "appearance": "transparent",
  "aria-expanded": true,
  "children": <div>
    Some header
  </div>,
  "className": "collapsible",
  "icon": <ChevronDownRegular
    className="collapsible"
  />,
  "onClick": [Function],
}
`;

exports[`CollapsibleComponentTest toggle from expanded to collapsed: Button props 2`] = `
{
  "appearance": "transparent",
  "aria-expanded": true,
  "children": <div>
    Some header
  </div>,
  "className": "collapsible",
  "icon": <ChevronDownRegular
    className="collapsible"
  />,
  "onClick": [Function],
}
`;

exports[`CollapsibleComponentTest toggle from expanded to collapsed: collapsed 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <mock-button
      appearance="transparent"
      aria-expanded="true"
      classname="collapsible"
      icon="[object Object]"
    >
      <div>
        Some header
      </div>
    </mock-button>
    <div
      class="collapsibleContent"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleComponentTest toggle from expanded to collapsed: expanded 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <mock-button
      appearance="transparent"
      aria-expanded="true"
      classname="collapsible"
      icon="[object Object]"
    >
      <div>
        Some header
      </div>
    </mock-button>
    <div
      class="collapsibleContent"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
